Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -173,6 +181,26 @@ public class PBXProjGenerator {
 
         let configs: [XCBuildConfiguration] = spec.configs.map { config in
             var buildSettings = spec.getTargetBuildSettings(target: target, config: config)
+            // Do not overwrite target's xcconfig values.
+            if let configPath = spec.targets.flatMap({ $0 == target ? $0.configFiles[config.name] : nil }).first {
```

Comment:
Can this be moved into SettingsBuilder. 
Also is there a reason this can't just be `if let configPath = target.configFiles[config.name]`. Maybe I'm missing something